moduleframework/helpers/openshift_helper.py

```diff
@@ -0,0 +1,268 @@
+        # dovecot     172.30.1.1:5000/myproject/dovecot     latest    15 minutes ago
+        # memcached   172.30.1.1:5000/myproject/memcached   latest    13 minutes ago
+
+        app_found = [x for x in oc_get.split('\n') if x.startswith(self.app_name)]
+        if app_found:
+            # If application exists in svc / dc / is namespace, then remove it
+            oc_delete = self.runHost("oc delete %s/%s" % (oc_service, self.app_name), ignore_status=True)
+
+    def _app_remove(self):
+        """
+        Function removes an application from all OpenShift namespaces like 'svc', 'dc', 'is'
+        """
+        if self._app_exists():
+            # TODO get info from oc status and delete relevat svc/dc/is
+            for ns in ['svc', 'dc', 'is']:
+                self._remove_apps_from_openshift_namespaces(ns)
+
+    def _create_app(self):
+        """
+        It creates an application in OpenShift environment
+        """
+        # Switching to system user
+        oc_new_app = self.runHost("oc new-app %s --name=%s" % (self.container_name,
+                                                               self.app_name),
+                                  ignore_status=True)
+        common.print_info(oc_new_app.stdout)
+        time.sleep(1)
+
+    def _verify_pod(self):
+        """
+        It verifies if an application POD is initiated and ready for testing
+        :return: False, application is not initiated during 10 seconds
+                 True, application is initiated and ready for testing
+        """
+        pod_initiated = False
+        for x in range(0, 20):
+            pod_state = self.runHost("oc get pods", ignore_status=True)
+            pod_state = pod_state.stdout.split('\n')
+            for pod in pod_state:
+                if pod.startswith(self.app_name):
+                    if "Running" in pod and "deploy" not in pod:
+                        pod_initiated = True
+                        break
+            time.sleep(1)
+            if pod_initiated:
+                break
+        return pod_initiated
+
+    def setUp(self):
+        """
+        It is called by child class and it is same methof as Avocado/Unittest has. It prepares environment
+        for docker testing
+        * start docker if not
+        * pull docker image
+        * setup environment from config
+        * run and store identification
+
+        :return: None
+        """
+        self.icontainer = self.get_url()
+
+    def _openshift_login(self, oc_ip="127.0.0.1", oc_user='developer', oc_passwd='developer', env=False):
+        """
+        It logins to an OpenShift environment on specific IP and under user and his password.
+        :param oc_ip: an IP where is an OpenShift environment running
+        :param oc_user: an username under which we can login to OpenShift environment
+        :param oc_passwd: a password for specific username
+        :param env:
+        :return:
+        """
+        if env:
```

Comment:
I probably don't understand this logic

Reply:
Does this make sense to you?
```env: is used for specification OpenShift IP, user and password, otherwise defaults are used```

Reply:
no, I still don't understand the reason for that variablee, since the values are being picked up from environment